### PR TITLE
fix: proxyUrl in incognito context

### PR DESCRIPTION
**Pull request by szmarczak**
Fixes https://github.com/apify/apify-js/issues/1195 a breaking change introduced by https://github.com/puppeteer/puppeteer/pull/7516

### CODE REVIEW

File: src/puppeteer/puppeteer-plugin.ts
Comment (szmarczak) [inline src/puppeteer/puppeteer-plugin.ts:50]: Is the `proxyUrl` always anonymized? If so the latter would be moot

Comment (szmarczak) [reply src/puppeteer/puppeteer-plugin.ts:50]: Ah it's not anonymized if there's no username & password.

**Review by mnmkng — APPROVED** [review]

File: src/puppeteer/puppeteer-plugin.ts
Comment (B4nan) [inline src/puppeteer/puppeteer-plugin.ts:20]: i guess the `anonymizedProxyUrl` comes from the index signature on `LaunchContext`, right? would be great to redesign the typings so we see actual options and their types, I believe it should be possible (but not in this PR)

**Review by B4nan — APPROVED** [review]

**Comment by B4nan** [comment]
please update the changelog before merging

(we really need to get this automated soon, it gets so tedious :])

**Comment by corford** [comment]
@szmarczak friendly prod (...unless we temporarily fork the repo we're blocked on this hotfix until it gets merged)

**Comment by B4nan** [comment]
We will merge this later today or tomorrow, and ship sdk 2.1 probably on wednesday or thursday with the updated version too.

**Comment by corford** [comment]
Ok, awesome - thanks!

**Comment by B4nan** [comment]
browser pool 2.0.4 stable is out with the fix (will get published in few minutes)

**Pull request merged**
